tce/src/tools/Environment.cc

```diff
@@ -1087,8 +1073,27 @@ std::vector<std::string> Environment::implementationTesterTemplatePaths() {
         string path = string(TCE_SRC_ROOT) + DS + "data" + DS + "hdb";
         paths.push_back(path);
     }
-    string path = string(TCE_INSTALLATION_ROOT) + string(INSTALLATION_DIR) 
-        + "data" + DS + "hdb";
+    string path = instPath() + string(INSTALLATION_DIR) + "data" + DS + "hdb";
     paths.push_back(path);
     return paths;
 }
+
+/**
+ * Returns the installation path for TCE
+ * It can be .deb unpacked to homedir (TCE_USER_INSTALL envvar should be set)
+ * or one created by 'make install'
+ *
+ * @return Path to TCE installation root
+ */
+string
+Environment::instPath() {
```

Comment:
This method belongs to the Application class which is a global access point for application-specific (runtime) data. Also, it should be initialized once (and cached) there so we don't call getenv and file exists all the time.